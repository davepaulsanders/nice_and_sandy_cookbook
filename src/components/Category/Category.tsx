import Wrapper from "../Wrapper/Wrapper"
import RecipeCard from "../RecipeCard/RecipeCard"
const Category = ({category, recipes, setRecipes, pinnedRecipes, setPinnedRecipes}: {category: string, recipes: any[], setPinnedRecipes: any, setRecipes: any, pinnedRecipes: any[]}) => {
	return (
		<section className="my-16 w-full">
		<p className="text-lightg font-bold text-4xl my-10 text-center">{category}</p>
		<Wrapper>
		{recipes.map(recp => (<RecipeCard {...recp} recipes={recipes} pinnedRecipes={pinnedRecipes} setRecipes={setRecipes} setPinnedRecipes={setPinnedRecipes}/>))}
		</Wrapper>
		</section>
	)
}

export default Category
