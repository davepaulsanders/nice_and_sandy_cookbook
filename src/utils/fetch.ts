export const fetchData = async (url: string) => {
	const resp = await fetch(url)
	if (resp.status !== 200) throw new Error(`Request failed with ${resp.status}`)
	const data = resp.json()
	return data
}
