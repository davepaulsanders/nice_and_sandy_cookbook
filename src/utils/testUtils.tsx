import RecipeCard from '../components/RecipeCard/RecipeCard';
import { ReactElement } from "react"
import { vi } from "vitest"
import { Recipe } from "../types/types"

const generateRecipes = () => {
const recipes: Recipe[] = []
for (let i = 0; i < 10; i++) {
	recipes.push({
	id:i + 1,
	img:"https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg",
	href:"https://www.feastingathome.com/vegan-fajitas/",
	label:`Vegan Fajitas ${i + 1}`,
	alt:"test",
	category:"ENTREE",
	category_id:1,
	is_pinned: false,
	})
}
return recipes
}
const generateRecipeCards = () => {
const recipeCards: ReactElement[] = []
for (let i = 0; i < 10; i++) {
	recipeCards.push(
	<RecipeCard
	id={i + 1}
	img="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
	href="https://www.feastingathome.com/vegan-fajitas/"
	label="Vegan Fajitas 2"
	alt="test"
	category="ENTREE"
	category_id={1}
	is_pinned= {false}
	recipes={generateRecipes()}
	setRecipes={vi.fn()}
	/>)
	}
	return recipeCards
}
export { generateRecipeCards, generateRecipes }
