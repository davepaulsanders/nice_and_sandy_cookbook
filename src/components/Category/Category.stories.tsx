import { vi } from "vitest"
import type { Meta, StoryObj } from '@storybook/react';
import Category from './Category';
 
import { generateRecipes } from "../../utils/testUtils"

const recipes = generateRecipes()

const meta: Meta<typeof Category> = {
  component: Category,
};
 
export default meta;

type Story = StoryObj<typeof Category>;
const mock = vi.fn()
// @ts-ignore
export const CategoryStory: Story = {args: {category: "Entree", recipes: recipes, setRecipes={mock}}}
