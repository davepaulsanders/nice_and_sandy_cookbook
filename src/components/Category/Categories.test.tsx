import Category from "./Category" 
import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import '@testing-library/dom'

import { generateRecipeCards } from "../../utils/testUtils"

const children = generateRecipeCards()

test('<Category /> renders', async () => {
	const { container } = render(<Category category="Entree" children={children}/>)
	expect(container).toMatchSnapshot()
})
