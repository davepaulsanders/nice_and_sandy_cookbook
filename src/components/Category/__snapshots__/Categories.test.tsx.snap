// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Category /> renders 1`] = `
<div>
  <section
    class="w-full"
  >
    <p
      class="text-lightg font-bold text-3xl my-10 text-center"
    >
      Entree
    </p>
    <div
      class="grid gap-4 grid-cols-2 sm:grid-cols-6 sm:mx-auto"
    />
  </section>
</div>
`;
